api/v1alpha2/baremetalcluster_types.go

```diff
@@ -28,6 +28,7 @@ const (
 
 // BareMetalClusterSpec defines the desired state of BareMetalCluster.
 type BareMetalClusterSpec struct {
+	APIEndpoint string `json:"apiEndpoint,omitempty"`
```

Comment:
BareMetalClusterStatus struct already contains APIEndpoints list, why not use it?
The BareMetalClusterSpec comment also implies the struct content.